src/RequestHandler/inprocess/inprocessapplication.h

```diff
@@ -57,6 +52,28 @@ class IN_PROCESS_APPLICATION : public APPLICATION
         IN_PROCESS_HANDLER* pInProcessHandler
     );
 
+    VOID
+    StopCallsIntoManaged(
+        VOID
+    )
+    {
+        AcquireSRWLockExclusive(&m_srwLock);
```

Comment:
what's the purpose of lock here? is volatile or interlockcomparedexchange not sufficient?